projects/plugins/jetpack/extensions/blocks/tiled-gallery/tiled-gallery.php

```diff
@@ -210,6 +267,645 @@ private static function is_squareish_layout( $attr ) {
+	 *
+	 * @param array  $image Image data array.
+	 * @param string $additional_styles Additional CSS styles.
+	 * @param string $border_radius_style Border radius CSS.
+	 * @param array  $attr Block attributes (optional, for link processing).
+	 * @return string Image HTML.
+	 */
+	private static function generate_image_html( $image, $additional_styles = '', $border_radius_style = '', $attr = array() ) {
+		$base_styles     = 'border:none;background-color:#0000001a;display:block;height:auto;max-width:100%;object-fit:cover;object-position:center;padding:0;';
+		$combined_styles = $base_styles . $additional_styles . $border_radius_style;
+
+		$img_html = sprintf(
+			'<img alt="%s" src="%s" style="%s" />',
+			esc_attr( $image['alt'] ),
+			esc_url( $image['url'] ),
+			$combined_styles
+		);
+
+		// Handle link settings for email
+		$link_to = ! empty( $attr['linkTo'] ) ? $attr['linkTo'] : 'none';
+		$href    = self::get_image_link_href( $image, $attr, $link_to );
+
+		if ( ! empty( $href ) ) {
+			return sprintf( '<a href="%s">%s</a>', esc_url( $href ), $img_html );
+		}
+
+		return $img_html;
+	}
+
+	/**
+	 * Get the href for an image based on link settings (used for email rendering).
+	 * Excludes custom links which email clients will replace with the image.
+	 *
+	 * @since $$next-version$$
+	 *
+	 * @param array  $image Image data array.
+	 * @param array  $attr Block attributes.
+	 * @param string $link_to Link setting.
+	 * @return string The href URL or empty string.
+	 */
+	private static function get_image_link_href( $image, $attr, $link_to ) {
+		switch ( $link_to ) {
+			case 'media':
+				return ! empty( $image['url'] ) ? $image['url'] : '';
+
+			case 'attachment':
+				// For email, we need to generate the attachment page URL from the image ID
+				if ( ! empty( $image['id'] ) ) {
+					$attachment_url = get_permalink( $image['id'] );
+					return $attachment_url ? $attachment_url : '';
+				}
+				return '';
+			default:
+				return '';
+		}
+	}
+
+	/**
+	 * Check if layout style is square or circle (squareish).
+	 *
+	 * @param string $layout_style The layout style.
+	 * @return bool True if squareish layout.
+	 */
+	private static function is_squareish_layout_style( $layout_style ) {
+		return in_array( $layout_style, array( 'square', 'circle' ), true );
+	}
+
+	/**
+	 * Create hierarchical chunks for square/circle layouts with larger items first.
+	 *
+	 * @param array $images Array of image data.
+	 * @param int   $columns Number of columns for the layout.
+	 * @return array Array of rows with different sized chunks.
+	 */
+	private static function create_hierarchical_chunks( $images, $columns ) {
+		$image_count = count( $images );
+		$chunks      = array();
+
+		if ( $image_count <= $columns ) {
+			// For column count or fewer, single row
+			$chunks[] = $images;
+		} else {
+			// Calculate remainder when dividing by columns
+			$remainder   = $image_count % $columns;
+			$start_index = 0;
+
+			if ( $remainder === 1 ) {
```

Comment:
This seems to be handling only the cases with 1 and 2 remainder, i.e. 2 or 3 columns. When I tested with 4 columns and 7 images, the hierarchy in the rendered email is flipped:

Editor:
<img width="770" height="448" alt="WWEay1BY0RUnNqVX" src="https://github.com/user-attachments/assets/1da023b9-d3c4-43cc-884d-785ca5944f1f" />

Email:
<img width="709" height="412" alt="wuOlWjCYgGegiXeq" src="https://github.com/user-attachments/assets/0ac5aa06-77d4-448f-9228-629da413994a" />